test/e2e/antreapolicy_test.go

```diff
@@ -1766,6 +1766,93 @@ func testAppliedToPerRule(t *testing.T) {
 	executeTests(t, testCase)
 }
 
+func testACNPClusterGroupServiceRefCreateAndUpdate(t *testing.T, data *TestData) {
+	svc1 := k8sUtils.BuildService("svc1", "x", 80, 80, map[string]string{"pod": "a"}, nil)
+	svc2 := k8sUtils.BuildService("svc2", "y", 80, 80, map[string]string{"pod": "b"}, nil)
+
+	cg1Name, cg2Name := "cg-svc1", "cg-svc2"
+	cgBuilder1 := &ClusterGroupSpecBuilder{}
+	cgBuilder1 = cgBuilder1.SetName(cg1Name)
+	cgBuilder1 = cgBuilder1.SetServiceReference("x", "svc1")
```

Comment:
The advantage of having the `SetX` functions return a pointer to the object is that we can have more concise test definitons: `cgBuilder1 = cgBuilder1.SetName(cg1Name).SetName(cg1Name)`

although in this case, it would be good to be able to simply write: `cgBuilder1 := NewClusterGroupSpecBuilder(cg1Name).SetServiceReference("x", "svc1")` to replace all 3 lines with a single 1

Reply:
Thanks for the suggestion. Will change this.